src/openms/source/DATASTRUCTURES/QTCluster.cpp

```diff
@@ -49,110 +49,116 @@ using std::iterator;
 
 namespace OpenMS
 {
-
-  QTCluster::QTCluster()
-  {
-  }
-
-  QTCluster::QTCluster(OpenMS::GridFeature* center_point, Size num_maps,
-                       double max_distance, bool use_IDs, Int x_coord, 
-                       Int y_coord) :
-    center_point_(center_point),
-    neighbors_(),
-    tmp_neighbors_(nullptr),
-    max_distance_(max_distance),
-    num_maps_(num_maps),
+  QTCluster::QTCluster(QTCluster::Data_* data, OpenMS::GridFeature* center_point, 
+                       Size num_maps, double max_distance,
+                       bool use_IDs, Int x_coord, Int y_coord, Size id) :
     quality_(0.0),
+    valid_(true),
     changed_(false),
     use_IDs_(use_IDs),
-    valid_(true),
     collect_annotations_(false),
     finalized_(true),
-    x_coord_(x_coord),
-    y_coord_(y_coord),
-    annotations_()
+    data_(data)
   {
+    data_->center_point_ = center_point;
```

Comment:
you could move all this into the `emplace_back` when the `Data_` is actually constructed.
I know you wanted to keep the interface similar to the old one, but there is no need for nostalgia in this code..

Reply:
Done